ui/jstests/test-facets.js

```diff
@@ -0,0 +1,59 @@
+define(['QUnit', 'jquery', 'facets', 'icheck'], function(QUnit, $, facets) {
+  'use strict';
+  var testQueryString = 'selected_facets=foo:bar';
+  QUnit.test(
+    'Test constructFacetQuery with given input element',
+    function(assert) {
+      var testData = $(
+        '<input data-facet-name="foo" type="checkbox" data-facet-value="bar">'
+      )[0];
+      var facetQuery = facets.constructFacetQuery(testData);
+      assert.equal(testQueryString, facetQuery);
+    }
+  );
+  QUnit.test(
+    'Test checkFacets with given input element',
+    function(assert) {
+      $(
+        '<input id="foobar" type="checkbox"' +
+          'class="icheck-11" data-facet-name="foo" data-facet-value="bar">'
+      ).appendTo($('body'));
+      $('#foobar').iCheck();
+      facets.checkFacets(testQueryString);
+      assert.ok($('#foobar').prop('checked'));
+      $('#foobar').remove();
```

Comment:
We could create some parent container div on setup then remove it on teardown, and everything associated with it would get removed automatically. But it's not a blocker for this issue